### DISCUSSION TITLE
ESP32 + mcp2515 + ide arduino

### DISCUSSION THREAD

**Discussion by leompa** [Open-Agriculture/AgIsoStack-plus-plus#262] [General]
Hello, I am reviewing the project to adapt it to ESP32 + MCP2515, but I want to do it from the Arduino ID and the truth is not a form.
There is a possibility or I must do it with Platform Io.
The idea is to make a VT that shows in any console the rotation of the axes of a machine.

From now on thank you very much and I loved finding an open project.

I am from Argentina.

**Comment by ad3154** [comment]
Hello! Currently we do not support the Arduino IDE, but this is a planned feature for the future at some point for both Teensy and ESP. It may take quite some time for us to get there, so for now Platform IO is perhaps the best if ESP32 is your target platform.

Best of luck with your VT project, and please let us know if you find any bugs!

**Comment by leompa** [comment]
Hello, perfect I will start to investigate the use of plasform io, the next question is can you use the mcp2515 + esp32 combination with your library. I'll be trying to run the VT example to learn. thank you so much.

**Reply by ad3154** [reply]
We do support the MCP2515 with the ESP32.

There may be some things to consider with the ESP32 platform, such as needing to embed the object pool's `.iop` file like so: https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-guides/build-system.html#embedding-binary-data

If you get stuck, @GwnDaan may be able to help.

On a side note, @GwnDaan we get inquiries like this with increasing frequency, maybe we should write some getting started documentation for ESP32 + MCP2515?

**Reply by GwnDaan** [reply]
> On a side note, @GwnDaan we get inquiries like this with increasing frequency, maybe we should write some getting started documentation for ESP32 + MCP2515?

That sounds like a plan, I created issue #263 for this
